### PR TITLE
Node.js CLI post

### PR DESCRIPTION
**Pull request by reverentgeek**
## THIS PR IS A BLOG POST ADDITION OR UPDATE ONLY
If you need to submit changes for documentation, plesae do so at [our new repo](https://github.com/okta/okta-developer-docs)

## Description:
- Describe your changes. This will most likely be a copy-paste of your commit
  message(s), which should be descriptive and informative.
- Is this PR related to a Monolith release? If so, which one?

### Resolves:

* [OKTA-XXXXXX](https://oktainc.atlassian.net/browse/OKTA-XXXXXX)

### CODE REVIEW

File: _source/_posts/2019-06-18-command-line-app-with-nodejs.md [hunk 1]
```diff
@@ -0,0 +1,545 @@
...
+
+Next, open the `package.json` file in the root of the project in your text editor. Change the `main` value to `bin/index.js`. Add a new key for `bin` with the following text.
+
+```json
+ "bin": {
+   "hello": "./bin/index.js"
+ }
+```
+
+> Note: Don't forget to add a comma after the previous value. All key/value pairs are separated by commas. Often the most common mistake when editing a JSON file is leaving out a comma.
```
Comment (dogeared) [inline _source/_posts/2019-06-18-command-line-app-with-nodejs.md]: nit: Often the most common mistake... reads a little awkwardly. Maybe just: The most common mistake... or: Often the mistake...

Not a show stopper...

File: _source/_posts/2019-06-18-command-line-app-with-nodejs.md [hunk 2]
```diff
@@ -0,0 +1,545 @@
...
+Next, click on the **Dashboard** link. At the top of the dashboard on the right side of the page you should find your Okta **Org URL**. Copy this value and put this in your `.env` file.
+
+{% img blog/command-line-apps-with-nodejs/okta-07-org-url.png alt:"Your Okta Org URL" width:"785" %}{: .center-image }
+
+### Update the Node.js Command Line Application
+
+To support the PKCE authentication flow, your CLI application needs a few more libraries. In this tutorial, you will use `hapi` to create a web server to handle the authentication callback. `dotenv` is used to read configuration settings from the `.env` file. You will use `open` to launch the default browser for login. And, `uuid` is used to help generate a unique private key to exchange with the authentication server.
+
+```bash
+npm install @hapi/hapi@18.3 dotenv@8.0 open@6.3 uuid@3.3
```
Comment (dogeared) [inline _source/_posts/2019-06-18-command-line-app-with-nodejs.md]: FYI, the okta-auth-js library supports pkce now. Maybe an update for future is to use okta-auth-js?

Not a show stopper...

File: _source/_posts/2019-06-18-command-line-app-with-nodejs.md [hunk 3]
```diff
@@ -0,0 +1,545 @@
...
+1. The `codeChallenge` private key is generated.
+1. The function opens the default browser and navigates to the authorization server.
+1. If not already logged in, the user must log in to the authentication server.
+1. Once authenticated, the browser is redirected to the `/callback` URL with a code.
+1. The callback handler uses the code and the `codeChallenge` value to request an authentication token.
+1. The authorization server verifies the code and `codeChallenge` values and replies with an authentication token.
+1. The function uses the returned token to call another API endpoint to retrieve account information.
+1. The token and account information are returned to the caller that invoked `executeAuthFlow`.
+
+{% img blog/command-line-apps-with-nodejs/cli-pkce-auth-flow.png alt:"CLI PKCE auth flow" width:"800" %}{: .center-image }
```
Comment (dogeared) [inline _source/_posts/2019-06-18-command-line-app-with-nodejs.md]: This image has one incorrect part of the flow:

after you receive the code, you hit the /token endpoint with the code + verifier NOT the code + challenge.

This needs to be updated.

File: _source/_posts/2019-06-18-command-line-app-with-nodejs.md (hunk 1 above)
Comment (reverentgeek) [reply _source/_posts/2019-06-18-command-line-app-with-nodejs.md]: Updated

File: _source/_posts/2019-06-18-command-line-app-with-nodejs.md (hunk 2 above)
Comment (reverentgeek) [reply _source/_posts/2019-06-18-command-line-app-with-nodejs.md]: I'll have to cover this another time, I think.

File: _source/_posts/2019-06-18-command-line-app-with-nodejs.md (hunk 3 above)
Comment (reverentgeek) [reply _source/_posts/2019-06-18-command-line-app-with-nodejs.md]: Updated image and description. Thank you!

**Pull request merged**